src/main/scala/org/ergoplatform/modifiers/history/Extension.scala

```diff
@@ -81,41 +78,31 @@ object Extension extends ApiCodecs {
   }
 }
 
-object ExtensionSerializer extends Serializer[Extension] {
-  val Delimiter: Array[Byte] = Array(0: Byte, Byte.MinValue)
+object ExtensionSerializer extends ScorexSerializer[Extension] {
 
-  override def toBytes(obj: Extension): Array[Byte] = {
-    val mandBytes = scorex.core.utils.concatBytes(obj.fields.map(f =>
-      Bytes.concat(f._1, Array(f._2.length.toByte), f._2)))
+  override def serialize(obj: Extension, w: Writer): Unit = {
+    w.putBytes(idToBytes(obj.headerId))
 
-    Bytes.concat(idToBytes(obj.headerId), mandBytes)
+    w.putUShort(obj.fields.size)
+    obj.fields.foreach { case (key, value) =>
+      w.putBytes(key)
+      w.putUByte(value.length)
+      w.putBytes(value)
+    }
   }
 
-  override def parseBytes(bytes: Array[Byte]): Try[Extension] = Try {
-    val totalLength = bytes.length
-
-    require(totalLength < Constants.extensionMaxSize)
-
-    @tailrec
-    def parseFields(pos: Int,
-                     acc: Seq[(Array[Byte], Array[Byte])]): Seq[(Array[Byte], Array[Byte])] = {
-      val keySize = Extension.FieldKeySize
-      if (pos == totalLength) {
-        // deserialization complete
-        acc.reverse
-      } else {
-        val key = bytes.slice(pos, pos + keySize)
-        val length: Byte = bytes(pos + keySize)
-        require(length >= 0, s"value size should not be negative, length = $length, pos = $pos")
-        require(length <= Extension.FieldValueMaxSize, "value size should be <= " + Extension.FieldValueMaxSize)
-        val value = bytes.slice(pos + keySize + 1, pos + keySize + 1 + length)
-        parseFields(pos + keySize + 1 + length, (key, value) +: acc)
-      }
+  override def parse(r: Reader): Extension = {
+    val startPosition = r.position
+    val headerId = bytesToId(r.getBytes(Constants.ModifierIdSize))
+    val fieldsSize = r.getUShort()
+    val fields = (1 to fieldsSize) map {_ =>
+      val key = r.getBytes(Extension.FieldKeySize)
+      val size = r.getUShort()
```

Comment:
during serialization you put UByte but here you get UShort. How does this works?